locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/networking/SharedLocationSecurityGroupCustomizer.java

```diff
@@ -76,6 +79,14 @@
     private RangeSet<Integer> udpPortRanges;
 
     /**
+     * Tested only on AWS only.
+     * It depends on the cloud provider and jclouds driver whether security group allows opening ICMP.
+     */
+    private Boolean openIcmp;
+
+    private Collection<SecurityGroup> updatedSecurityGroups;
```

Comment:
Delete this and `getUpdatedSecurityGroups` and add a new method to the class (not sure of the best name for it, maybe simply `doCustomize`?) that performs the customisation and returns the new security groups. `customize(JcloudsLocation, ComputeService, JcloudsMachineLocation)` should call this new method and discard the result.